core-common/src/main/java/eu/arrowhead/common/verifier/NetworkAddressVerifier.java

```diff
@@ -0,0 +1,183 @@
+	// methods
+	
+	//-------------------------------------------------------------------------------------------------
+	public void verify(final String address) throws InvalidParameterException { //TODO junit
+		logger.debug("verify started...");
+		
+		if (Utilities.isEmpty(address)) {
+			throw new InvalidParameterException("network address is empty");
+		}
+		
+		final String candidate = address.toLowerCase().trim();
+		if (ipv4Pattern.matcher(candidate).matches()) {
+			verifyIPV4(candidate);
+			
+		} else if (ipv6Pattern.matcher(candidate).matches()) {
+			verifyIPV6(candidate);
+			
+		} else if (domainNamePattern.matcher(candidate).matches()) {
+			verifyDomainName(candidate);
+		} else {
+			verifyNoType(candidate);			
+		}		
+	}
+	
+	//=================================================================================================
+	// assistant methods
+	
+	//-------------------------------------------------------------------------------------------------
+	private void verifyIPV4(final String candidate) {
+		logger.debug("verifyIPV4 started...");
+		
+		if (!allowSelfAddressing) {			
+			// Filter out IP placeholder(default route) (0.0.0.0) and loopback (127.0.0.0 - 127.255.255.255)
+			if (candidate.equalsIgnoreCase(IPV4_PLACEHOLDER) || candidate.startsWith(IPV4_LOOPBACK_1ST_OCTET)) {
+				throw new InvalidParameterException(candidate + " ipv4 network address is invalid: self-addressing is disabled");
+			}
+		}
+		
+		if (!allowNonRoutableAddressing) {
+			// Filter out APIPA (Automatic Private IP Address: 169.254.?.?)
+			if (candidate.startsWith(IPV4_APIPA_1ST_AND_2ND_OCTET)) {
+				throw new InvalidParameterException(candidate + " ipv4 network address is invalid: non-routable-addressing is disabled");
+			}
+		}
+		
+		// Filter out local broadcast (255.255.255.255)
+		if (candidate.equalsIgnoreCase(IPV4_LOCAL_BROADCAST)) {
+			throw new InvalidParameterException(candidate + " ipv4 network address is invalid: local broadcast address is denied");
+		}
+		
+		// Could not filter out directed broadcast (cannot determine it without the subnet mask)
+
+		// Filter out multicast (Class D: 224.0.0.0 - 239.255.255.255)
+		final String[] octets = candidate.split("\\.");
+		final Integer firstOctet = Integer.valueOf(octets[0]);
+		if (firstOctet >= IPV4_MULTICAST_1ST_OCTET_START && firstOctet <= IPV4_MULTICAST_1ST_OCTET_END) {
+			throw new InvalidParameterException(candidate + " ipv4 network address is invalid: multicast addresses are denied");
+		}		
+		
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	private void verifyIPV6(final String candidate) {
+		logger.debug("verifyIPV6 started...");
+		
+		if (!allowSelfAddressing) {
+			// Filter out unspecified address (0:0:0:0:0:0:0:0) and loopback address (0:0:0:0:0:0:0:1)			
+			if (candidate.equalsIgnoreCase(IPV6_UNSPECIFIED) || candidate.equalsIgnoreCase(IPV6_LOOPBACK)) {
```

Comment:
I think unspecified address is invalid regardless of the allowSelfAddressing flag